cmd/yggd/main.go

```diff
@@ -120,6 +121,35 @@ func setupClientID() error {
 	return nil
 }
 
+// setupFactsFile creates a canonical facts file if it doesn’t exist
+func setupFactsFile() error {
+	if config.DefaultConfig.FactsFile == "" {
+		return nil
+	}
+
+	_, err := os.Stat(config.DefaultConfig.FactsFile)
+	if err == nil {
+		return nil
+	}
+	if !os.IsNotExist(err) {
+		return fmt.Errorf("cannot stat file '%v': %v", config.DefaultConfig.FactsFile, err)
+	}
+
+	dir := filepath.Dir(config.DefaultConfig.FactsFile)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("cannot create directory '%v': %v", dir, err)
+	}
+
+	facts := make(map[string]interface{})
+	data, _ := json.Marshal(facts)
+	if err := os.WriteFile(config.DefaultConfig.FactsFile, data, 0644); err != nil {
+		return fmt.Errorf("cannot create facts file '%v': %v", config.DefaultConfig.FactsFile, err)
+	}
+
+	log.Infof("facts file not found, created")
```

Comment:
Hey, something I just thought of that might be useful here. Can we print the file name as part of this info message?
```suggestion
	log.Infof("facts file not found, created %v", config.DefaultConfig.FactsFile)
```

Reply:
Great idea!